### PR TITLE
Final Patches for RCD related devices.

### LINKED ISSUE #41114: Mounted RCD does not work on chasms/bohbombs/lava
Reproduction: stick mounted RCD on an APLU, select it and choose construct mode, click on a chasm, watch as nothing happens. Click on basalt and see it build a wall.

Hand-held RCDs, including borg's, work as expected.

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
Final closing act for #74527, some more bugs to squash.

1. Fixes broken sprite for rack in RCD UI
Screenshot (153)

2. Fixes #41114 mounted RCD can lay plating on chasm/ any open turf again

3. Removes unused turf argument inside `try_clean()`, `try_wallmount()`, & `try_decon()` procs because well it was unused

## Changelog
:cl:
refactor: remove unused turf var inside wallmount procs
fix: broken rack sprite inside RCD UI
fix: mounted RCD now lays plating over chasms and open turfs
/:cl:

### CODE REVIEW

File: code/game/objects/items/rcd/RHD.dm [hunk 1]
```diff
@@ -201,18 +201,18 @@
 	return data
 
 /obj/item/construction/proc/toggle_silo(mob/user)
+	. = TRUE
```
Comment (Mothblocks) [inline code/game/objects/items/rcd/RHD.dm:204]: Why did you change this? This is worse than the code before it.

File: code/modules/vehicles/mecha/equipment/tools/work_tools.dm [hunk 2]
```diff
@@ -303,18 +303,18 @@
...
-					return
-				S.PlaceOnTop(/turf/open/floor/plating, flags = CHANGETURF_INHERIT_AIR)
-			else if(isfloorturf(target))
+			if(isfloorturf(target))
 				var/turf/open/floor/F = target
 				to_chat(source, "[icon2html(src, source)][span_notice("Building Wall...")]")
 				if(!do_after_cooldown(F, source))
 					return
 				F.PlaceOnTop(/turf/closed/wall)
+			else if(isopenturf(target))
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/equipment/tools/work_tools.dm:312]: Why change to isopenturf? This doesn't look better than isspaceturf considering we're casting to space turf.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/game/objects/items/rcd/RHD.dm (hunk 1 above)
Comment (SyncIt21) [reply code/game/objects/items/rcd/RHD.dm:204]: I made an wrong assumption previously. The return value of this proc should always be true even if it doesn't succeed because ui_act() will check that to determine if the action was handled or not and so it's not important if the action was successful or not

This is the parent proc for all rcd like devices 
https://github.com/tgstation/tgstation/blob/e16588b682a8a1516790d00ff85b305fcc62c11d/code/game/objects/items/rcd/RHD.dm#L217-L223

So since toggle_silo() now always returns true take the rcd for example.  It can now early return here
https://github.com/tgstation/tgstation/blob/e16588b682a8a1516790d00ff85b305fcc62c11d/code/game/objects/items/rcd/RCD.dm#L463-466.

If it did return false instead when the silo link failed then it would waste time going through the remaining if conditions

File: code/modules/vehicles/mecha/equipment/tools/work_tools.dm (hunk 2 above)
Comment (SyncIt21) [reply code/modules/vehicles/mecha/equipment/tools/work_tools.dm:312]: To cover all edge cases? I'm not sure I'll make it just  isspaceturf() then

Comment (SyncIt21) [reply code/modules/vehicles/mecha/equipment/tools/work_tools.dm:312]: Ok it has to be isopenturf() becuase chasm isn't a subtype of space turf
https://github.com/tgstation/tgstation/blob/e16588b682a8a1516790d00ff85b305fcc62c11d/code/game/turfs/open/chasm.dm#L1-L3

File: code/game/objects/items/rcd/RHD.dm (hunk 1 above)
Comment (Mothblocks) [reply code/game/objects/items/rcd/RHD.dm]: `ui_act`'s return value isn't if it succeeded, it's if the UI should update. Sometimes it's reasonable to return FALSE, preferences does that a lot.
